src/lib.rs

```diff
@@ -704,10 +707,9 @@ pub fn init<UserData: 'static>(mut callback: Option<ErrorCallback<UserData>>) ->
     extern "C" fn glfw_terminate() {
         unsafe {
             ffi::glfwTerminate();
+            INIT = Once::new();
```

Comment:
This write on share data happens without synchronization and is very possibly not sound. `static mut` has many issues prompting multiple proposals for deprecation over the years.

Furthermore, except in `cfg!(feature = "terminate_manually")` this resets the init check. The result is either erroring regardless due to `glfwInit` returning false or registering a duplicate `glfw_terminate` to be called with `atexit`.